added tenacity error-retry

Magic73 opened the pull request:
This will fix [Errno 113] Host is unreachable, in the case of a WiFi problem.

Original issue description: https://github.com/slyglif/powerwall3mqtt/issues/51

Magic73 committed:
Update __init__.py

added exponential backoff retries.

Magic73 committed:
Update config.yaml

Magic73 committed:
Update __init__.py

Magic73 committed:
Update CHANGELOG.md

Magic73 committed:
Update __init__.py

Magic73 committed:
Update __init__.py

Magic73 committed:
Update config.yaml

Magic73 committed:
Update __init__.py

Magic73 committed:
Update config.yaml

Magic73 committed:
Update config.yaml

Magic73 committed:
Update __init__.py

Magic73 committed:
Update Dockerfile

Magic73 committed:
Update config.yaml

Magic73 committed:
Update Dockerfile

Magic73 committed:
Update config.yaml

Magic73 committed:
Update __init__.py

Magic73 committed:
Update config.yaml

slyglif commented on Dockerfile:
```diff
@@ -2,8 +2,8 @@ ARG BUILD_FROM
 # Copy data for add-on
-RUN apk add --no-cache python3 py3-pip py3-pyaml py3-cachetools py3-requests py3-protobuf tini
-RUN pip3 install --break-system-packages paho-mqtt
+RUN apk add --no-cache python3 py3-pip py3-pyaml py3-cachetools py3-requests py3-protobuf tini 
```
Nit: remove trailing space

slyglif commented on app/pytedapi/__init__.py:
```diff
@@ -209,13 +210,25 @@ def request(self, path, force=False):
+            except (requests.exceptions.ConnectionError, requests.exceptions.Timeout) as e:
+                raise exceptions.TEDAPIException("Failed to reach host after multiple retries") from e
+    
+        self.check_http_response(r)
```
This and the next line should stay inside the `with self._api_lock block` (so indent one level).

slyglif commented on app/pytedapi/__init__.py:
```diff
@@ -238,15 +251,28 @@ def post(self, path, force=False, headers=None, data=None):
+            except (requests.exceptions.ConnectionError, requests.exceptions.Timeout) as e:
+                raise exceptions.TEDAPIException("Failed to reach host after multiple retries") from e
+    
+        self.check_http_response(r)
```
This and the next 2 lines should stay inside the `with self._api_lock` block (so indent one level)

slyglif reviewed (changes requested):
Sorry for the delay on this, but I finally got time to go through and review the PR.  I did test and it seems to work, so once the feedback is addressed I'll merge and cut the new release.

Magic73 reviewed (commented):
Ok for me, and thank you for your plugin!

Magic73 committed:
Adjust timeout and fix indentation in __init__.py

https://github.com/slyglif/powerwall3mqtt/pull/55#discussion_r2595245320

https://github.com/slyglif/powerwall3mqtt/pull/55#discussion_r2595245632

Magic73 committed:
Update project name and version in config.yaml

https://github.com/slyglif/powerwall3mqtt/pull/55#discussion_r2595245848
https://github.com/slyglif/powerwall3mqtt/pull/55#discussion_r2595246004

Magic73 committed:
Remove 0.5.3 entry from CHANGELOG.md

https://github.com/slyglif/powerwall3mqtt/pull/55#discussion_r2595247864

Magic73 committed:
Merge branch 'slyglif:main' into main

slyglif commented:
Sorry, a couple of the issues are still there.  I reopened the discussions on each of the 2 left.

Magic73 committed:
Update Dockerfile

Magic73 committed:
Fix indentation for check_http_response call

Magic73 commented:
:)
